src/actions.js

```diff
@@ -878,22 +889,56 @@ export const networkSettingsUpdate = (customNetworkRequest) => ({
  *   walletServiceWsUrl?: string
  * }} customNetwork Settings to persist
  */
-export const networkSettingsUpdateSuccess = (customNetwork) => ({
-  type: types.NETWORKSETTINGS_UPDATE_SUCCESS,
+export const networkSettingsUpdateState = (customNetwork) => ({
+  type: types.NETWORKSETTINGS_UPDATE_STATE,
+  payload: customNetwork,
+});
+
+/**
+ * Emits the custom network settings to persist in the app storage and update the redux store.
+ * @param {{
+ *   stage: string,
+ *   network: string,
+ *   nodeUrl: string,
+ *   explorerUrl: string,
+ *   explorerServiceUrl: string,
+ *   walletServiceUrl?: string
+ *   walletServiceWsUrl?: string
+ * }} customNetwork Settings to persist
+ */
+export const networkSettingsPersistStore = (customNetwork) => ({
+  type: types.NETWORKSETTINGS_PERSIST_STORE,
   payload: customNetwork,
 });
 
+/**
+ * Emits the waiting signal after persist the custom network.
+ * It means the wallet will reload.
+ */
+export const networkSettingsUpdateWaiting = () => ({
+  type: types.NETWORKSETTINGS_UPDATE_WAITING,
+});
+
+/**
+ * Emits the success signal after wallet reloads.
+ * It servers as hook for the frontend to provide feedback for the user.
```

Comment:
```suggestion
 * Action indicating that the network settings update was successful
 * This serves as a hook for the frontend to provide feedback to the user.
```

Reply:
The idea here (and in the previous comments) is that the action by itself does not emit anything, somewhere else is dispatching this action